src/streaming/StreamProcessor.js

```diff
@@ -271,13 +271,15 @@ function StreamProcessor(config) {
     function onBufferCleared(e) {
         if (e.streamId !== streamInfo.id || e.mediaType !== type) return;
 
-        if (e.unintended) {
-            // There was an unintended buffer remove, probably creating a gap in the buffer, remove every saved request
-            fragmentModel.removeExecutedRequestsAfterTime(e.from);
-        } else {
-            fragmentModel.syncExecutedRequestsWithBufferedRange(
-                bufferController.getBuffer().getAllBufferRanges(),
-                streamInfo.duration);
+        // Remove executed requests not buffered anymore
+        fragmentModel.syncExecutedRequestsWithBufferedRange(
+            bufferController.getBuffer().getAllBufferRanges(),
+            streamInfo.duration);
+
+        // If buffer removed ahead current time (QuotaExceededError or automatic buffer pruning) then adjust current index handler time
+        if (e.from > playbackController.getTime()) {
+            setIndexHandlerTime(e.from);
```

Comment:
I think the code flow can be irritating here. We set the time of the indexHandler and then use it in the Streamprocessor when setting finding the next request. Can we keep the time in the streamProcessor and avoid this "roundtrip"?

Reply:
agree, replaced by bufferingTime in StreamProcessor